File: libgdx_sokoban/core/src/com/nopalsoft/sokoban/game_objects/EndPoint.java
package com.nopalsoft.sokoban.game_objects;

import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.graphics.g2d.TextureAtlas.AtlasRegion;
import com.nopalsoft.sokoban.Assets;

public class EndPoint extends Tile {
    int numColors;

    AtlasRegion keyFrame;

    public EndPoint(int position, String color) {
        super(position);

        switch (color) {
            case "brown":
                numColors = Box.COLOR_BROWN;
                break;
            case "gray":
                numColors = Box.COLOR_GRAY;
                break;
            case "purple":
                numColors = Box.COLOR_PURPLE;
                break;
            case "blue":
                numColors = Box.COLOR_BLUE;
                break;
            case "black":
                numColors = Box.COLOR_BLACK;
                break;
            case "beige":
                numColors = Box.COLOR_BEIGE;
                break;
            case "yellow":
                numColors = Box.COLOR_YELLOW;
                break;
            case "red":
                numColors = Box.COLOR_RED;
                break;
        }

        setTextureColor(numColors);
    }

    private void setTextureColor(int numColor) {
        switch (numColor) {
            case Box.COLOR_BEIGE:
                keyFrame = Assets.endPointBeige;
                break;

            case Box.COLOR_BLACK:
                keyFrame = Assets.endPointBlack;
                break;

            case Box.COLOR_BLUE:
                keyFrame = Assets.endPointBlue;
                break;

            case Box.COLOR_BROWN:
                keyFrame = Assets.endPointBrown;
                break;

            case Box.COLOR_GRAY:
                keyFrame = Assets.endPointGray;
                break;

            case Box.COLOR_RED:
                keyFrame = Assets.endPointRed;
                break;

            case Box.COLOR_YELLOW:
                keyFrame = Assets.endPointYellow;
                break;

            case Box.COLOR_PURPLE:
                keyFrame = Assets.endPointPurple;
                break;

        }
    }

    @Override
    public void draw(Batch batch, float parentAlpha) {
        batch.draw(keyFrame, getX(), getY(), SIZE, SIZE);
    }
}

File: libgdx_sokoban/core/src/com/nopalsoft/sokoban/game_objects/EndPoint.kt
package com.nopalsoft.sokoban.game_objects

import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.graphics.g2d.TextureAtlas.AtlasRegion
import com.nopalsoft.sokoban.Assets

class EndPoint(position: Int, color: String) : Tile(position) {
    val numColors = when (color) {
        "brown" -> Box.COLOR_BROWN
        "gray" -> Box.COLOR_GRAY
        "purple" -> Box.COLOR_PURPLE
        "blue" -> Box.COLOR_BLUE
        "black" -> Box.COLOR_BLACK
        "beige" -> Box.COLOR_BEIGE
        "yellow" -> Box.COLOR_YELLOW
        "red" -> Box.COLOR_RED
        else -> {
            Box.COLOR_BROWN
        }
    }
    private var keyFrame: AtlasRegion? = null

    init {
        setTextureColor(numColors)
    }

    private fun setTextureColor(numColor: Int) {
        keyFrame = when (numColor) {
            Box.COLOR_BEIGE -> Assets.endPointBeige
            Box.COLOR_BLACK -> Assets.endPointBlack
            Box.COLOR_BLUE -> Assets.endPointBlue
            Box.COLOR_BROWN -> Assets.endPointBrown
            Box.COLOR_GRAY -> Assets.endPointGray
            Box.COLOR_RED -> Assets.endPointRed
            Box.COLOR_YELLOW -> Assets.endPointYellow
            Box.COLOR_PURPLE -> Assets.endPointPurple
            else -> {
                Assets.endPointBrown
            }
        }
    }

    override fun draw(batch: Batch, parentAlpha: Float) {
        batch.draw(keyFrame, x, y, size, size)
    }
}